test/remote/gateways/remote_card_connect_test.rb

```diff
@@ -0,0 +1,190 @@
+require 'test_helper'
+
+class RemoteCardConnectTest < Test::Unit::TestCase
+  def setup
+    @gateway = CardConnectGateway.new(fixtures(:card_connect))
+
+    @amount = 100
+    @credit_card = credit_card('4788250000121443')
+    @declined_card = credit_card('4387751111111053')
+    @options = {
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+    @check = check(routing_number: '053000196')
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'Approval', response.message
+  end
+
+  def test_successful_purchase_with_more_options
+    options = {
+      order_id: '1',
+      ip: "127.0.0.1",
+      email: "joe@example.com",
+      po_number: '5FSD4',
+      tax_amount: '50',
+      freight_amount: '29',
+      duty_amount: '67',
+      order_date: '20170507',
+      ship_from_date: '20877',
+      items: [
+        {
+          line_no: '1',
+          material: 'MATERIAL-1',
+          description: 'DESCRIPTION-1',
+          upc: 'UPC-1',
+          quantity: '1000',
+          uom: 'CS',
+          unit_cost: '900',
+          net_amnt: '150',
+          tax_amnt: '117',
+          disc_amnt: '0'
+        },
+        {
+          line_no: '2',
+          material: 'MATERIAL-2',
+          description: 'DESCRIPTION-2',
+          upc: 'UPC-1',
+          quantity: '2000',
+          uom: 'CS',
+          unit_cost: '450',
+          net_amnt: '300',
+          tax_amnt: '117',
+          disc_amnt: '0'
+        }
+      ]
+    }
+
+    response = @gateway.purchase(@amount, @credit_card, options)
+    assert_success response
+    assert_equal 'Approval', response.message
+  end
+
+  def test_successful_purchase_3DS
+    three_ds_options = @options.merge(
+      secure_flag: 'se3453',
+      secure_value: '233frdf',
+      secure_xid: '334ef34'
+    )
+    response = @gateway.purchase(@amount, @credit_card, three_ds_options)
+    assert_success response
+    assert_equal 'Approval', response.message
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Insufficient funds', response.message
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount, auth.authorization)
+    assert_success capture
+    assert_equal 'Approval Queued for Capture' || 'Approval Accepted', capture.message
```

Comment:
This will get executed as

    assert_equal 'Approval Queued for Capture', capture.message

since 'Approved Queue for Capture' is truthy. I think you meant something closer to

    assert_includes ['Approval Queued for Capture', 'Approval Accepted'], capture.message

Reply:
I always do that. Thanks for the catch

Reply:
It's doubly tricky for me to catch these, because Python can handle variants of this in certain situations. E.g., `if 0 < x < 5` actually does what it looks like it ought to do. So I always have to think a second when I see these in Ruby.